doc/apiary/v2/fiware-ngsiv2-reference.apib

```diff
@@ -37,6 +37,112 @@ It is recommended to follow the “url” link values,
+* A collection of properties, one per entity attribute. Property values can be single values or they can include,
+  `type` and properties representing metadata as well. Property values which correspond to the primitive JSON datatypes
+  "string", "number" and "boolean" and don't include metadadta are rendered as regular JSON values. Property values
+  which specify a custom type or include metadata are rendered as JSON objects with the following properties:
+  * `value`: for the attribute value, rendered as a regular JSON value depending on the type. Attributes with an
+    user-defined informative types are rendered as JSON strings.
+  * `type`: a JSON string which represents the attribute type provided at attrinute creation time.
+  * A collection of properties one per metadata field associated to the attribute value. Metadata properties follows the
+    same representation rules that applies to attributes properties (except that a metadata
+    cannot have metadata properties).
+
+
+ the entity creator (or was implicitely set if the creator doesn't specified type at creation date).
+
+Canonical format description (TBD).
+
+### Predefined types
+
+All attribute types not included in the list below are considered *user-defined informative* types. The values of
+user-defined informative types are represented by JSON string.
+
+* `number` (or `Number`):  identifies attribute meaning numbers, either integer or float (see https://developer.mozilla.org/en-US/docs/Glossary/Number).
+They are associated to JSON number type. They can be used in greater than, lesser than, greater or equal,
+lesser or equal and range filters.
+
+```
+"temperature": 32.5
+```
+
+* `string` (or `String`), which represents text string. They are represented by JSON strings.
+
+```
+"msg": "Hello world"
+```
+
+* `boolean` (or `Boolean`), which represent a logical value (either true or false). They are represented by JSON booleans.
+
+```
+"active": true
+```
+
+* `date` (or `Date`):  identifies attribute meaning a date, in ISO8601 format. They are represented by JSON strings. They
+can be used in greater than, lesser than, greater or equal, lesser or equal and range filters.
+
+```
+"timestamp": "2017-06-17T07:21:24.238Z"
+```
+
+
+* `geo:point`: identifies an attribute meaning a geolocation. They are represented by JSON strings. An
+entity can only have one `geo:point` attribute.
+
+```
+{
+  ...
+  "location": {
+    "value": "41.3763726, 2.1864475,14",
+    "type": "geo:point"
+  }
+}
+```
+
+At creation or updating time, the type has to be coherent with the value of the attribute, e.g the following
+will result in an error response:
+
+```
+{
+  "value": "23.5", -> value should be: 23.5
+  "type": "number"
+}
+
+
+{
+  "value": 23.5,   -> value should be: "23.5"
+  "type": "string"
+}
+
+{
+  "value": "23.5", -> value should be: "23.5" (user defined types always use JSON strings as value)
```

Comment:
This is related with my previous comment "This is yet under discussion". My feeling is that we haven reached a global consensus about this yet, so probably it is a good idea (for not blocking the PR more than needed) or marking this part with "TBD" in the text and defer it for a new PR.